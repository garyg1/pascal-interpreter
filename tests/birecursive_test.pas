program birecursive_test;

function factorialB(n: integer): integer;
begin
    writeln('B called');
    if n = 0 then 
        factorialB := 1
    else begin
        factorialB := n * factorialA(n - 1);
    end; 
end;

function factorialA(n: integer): integer;
begin
    writeln('A called');
    if n = 0 then 
        factorialA := 1
    else begin
        factorialA := n * factorialB(n - 1);
    end;
end;

begin
    writeln(factorialA(6));
    writeln(factorialA(5));
    writeln(factorialB(6));
    writeln(factorialB(5));
end.