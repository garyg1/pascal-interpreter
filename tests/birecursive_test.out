A called
B called
A called
B called
A called
B called
A called
720
A called
B called
A called
B called
A called
B called
120
B called
A called
B called
A called
B called
A called
B called
720
B called
A called
B called
A called
B called
A called
120
